### PR TITLE
feat: improve-remediation

### PR DESCRIPTION
**Pull request by robcresswell**
This patch is a rewrite of the remediation implementation. It makes several changes:

- Remove the reliance on `inspect`, and instead get top level deps by parsing the manifest file. This should be significantly quicker than a pip install, making the fix process much faster for larger projects
- Change the parser to attempt to better preserve package casing, version comparators, and extras (comments, markers, etc)
- Update tests and add new tests. In a couple of cases, this actually involves fixing the test fixture, as the fixture itself was making changes such as package name casing, adding extra new lines to the manifest etc.

### CODE REVIEW

File: test/remediation.spec.ts [hunk 1]
```diff
@@ -1,115 +1,181 @@
...
-    },
-    {}
-  );
-}
 
 describe('remediation', () => {
-  beforeEach(() => {
-    activateVirtualenv('remediation');
+  // const manifests = { 'requirements.txt': 'Jinja==2.7.2' };
+  // const upgrades = { 'jinja@2.7.2': { upgradeTo: 'jinja@2.8.1' } };
```
Comment (gitphill) [inline test/remediation.spec.ts]: unused?

Comment (robcresswell) [reply test/remediation.spec.ts]: Oops! Yes, will remove.

File: lib/index.ts [hunk 2]
```diff
@@ -78,18 +76,142 @@ export async function applyRemediationToManifests(
...
+      // Check if we have an upgrade; if we do, replace the version string with
+      // the upgrade, but keep the rest of the content
+      const upgrade = upgrades[`${name.toLowerCase()}@${version}`];
 
-  applyUpgrades(manifests, upgrades, topLevelDeps);
+      if (!upgrade) {
+        return originalText;
+      }
+
+      const newVersion = upgrade.upgradeTo.split('@')[1];
```
Comment (gitphill) [inline lib/index.ts]: are we sure the package name will never have a '@' in the name?

Comment (robcresswell) [reply lib/index.ts]: I believe so. See

https://pypi.org/search/?q=%22s%22&o= vs https://pypi.org/search/?q=%22%40%22&o=

**Pull request merged**